pydantic/json_schema.py

```diff
@@ -1479,44 +1532,18 @@ def model_schema(self, schema: core_schema.ModelSchema) -> JsonSchemaValue:
             if root_json_schema_extra:
                 json_schema_extra = root_json_schema_extra
 
-        json_schema = self._update_class_schema(json_schema, title, config.get('extra', None), cls, json_schema_extra)
-
-        return json_schema
-
-    def _update_class_schema(
-        self,
-        json_schema: JsonSchemaValue,
-        title: str | None,
-        extra: Literal['allow', 'ignore', 'forbid'] | None,
-        cls: type[Any],
-        json_schema_extra: JsonDict | JsonSchemaExtraCallable | None,
-    ) -> JsonSchemaValue:
-        if '$ref' in json_schema:
-            schema_to_update = self.get_schema_from_definitions(JsonRef(json_schema['$ref'])) or json_schema
-        else:
-            schema_to_update = json_schema
-
-        if title is not None:
-            # referenced_schema['title'] = title
-            schema_to_update.setdefault('title', title)
-
-        if 'additionalProperties' not in schema_to_update:
-            if extra == 'allow':
-                schema_to_update['additionalProperties'] = True
-            elif extra == 'forbid':
-                schema_to_update['additionalProperties'] = False
-
         if isinstance(json_schema_extra, (staticmethod, classmethod)):
             # In older versions of python, this is necessary to ensure staticmethod/classmethods are callable
             json_schema_extra = json_schema_extra.__get__(cls)
```

Comment:
```suggestion
            json_schema_extra = cast(JsonSchemaExtraCallable, json_schema_extra.__get__(cls))
```

Fixes the `FIXME` comment below

Reply:
Hmm, still having issues even with this change.

Reply:
Ha, `cast(Callable[..., None])` should work, let's not fight the type checker as we make the `inspect.signature` check ourselves

Reply:
We've discussed - just going to keep the type ignores here.